src/System.Collections.Immutable/src/System.Collections.Immutable.csproj

```diff
@@ -43,7 +43,7 @@
     <Compile Include="System\Collections\Generic\ISortKeyCollection.cs" />
     <Compile Include="System\Collections\Immutable\AllocFreeConcurrentStack.cs" />
     <Compile Include="System\Collections\Immutable\DictionaryEnumerator.cs" />
-    <Compile Include="System\Collections\Immutable\DisposableEnumeratorAdapter`2.cs" />
+    <Compile Include="System\Collections\Immutable\DisposableEnumeratorAdapter.Generic.cs" />
```

Comment:
I don't recall whether StyleCop runs on this code base any more, but back when it was written, StyleCop drove some of these filename conventions. I also find it very useful to have filenames based on the classes they contain, so cramming the number directly next to the name with no delimiter isn't something I care too much for, personally.
I'd rather see the bugs in source server mentioned in #8608 fixed. 

If we are to take a fix like this PR though, I'd rather see underscores and the arity preserved as @stoub proposes rather than ".generic." added to these files.